confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceSpaceHelpers.java

```diff
@@ -0,0 +1,195 @@
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.xwiki.contrib.confluence.filter.internal.input;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Set;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import javax.inject.Singleton;
+
+import org.xwiki.component.annotation.Component;
+import org.xwiki.contrib.confluence.filter.input.ConfluenceInputProperties;
+import org.xwiki.model.EntityType;
+import org.xwiki.model.reference.DocumentReference;
+import org.xwiki.model.reference.EntityReference;
+import org.xwiki.model.reference.EntityReferenceResolver;
+import org.xwiki.model.reference.SpaceReference;
+import org.xwiki.model.reference.WikiReference;
+import org.xwiki.query.Query;
+import org.xwiki.query.QueryException;
+import org.xwiki.query.QueryManager;
+
+import com.xpn.xwiki.XWiki;
+import com.xpn.xwiki.XWikiContext;
+import com.xpn.xwiki.XWikiException;
+import com.xpn.xwiki.doc.XWikiDocument;
+import com.xpn.xwiki.objects.BaseObject;
+
+/**
+ * Provides utility methods for managing and validating Confluence spaces within XWiki during a Confluence import.
+ *
+ * @version $Id$
+ * @since 9.88.5
+ */
+@Component (roles = ConfluenceSpaceHelpers.class)
+@Singleton
+public class ConfluenceSpaceHelpers
+{
+    private final ThreadLocal<ConfluenceInputProperties> properties = new ThreadLocal<>();
+
+    @Inject
+    private Provider<XWikiContext> contextProvider;
+
+    @Inject
+    private EntityReferenceResolver<String> entityReferenceResolver;
+
+    @Inject
+    private QueryManager queryManager;
+
+    /**
+     * Sets the Confluence input properties.
+     * @param properties the Confluence input properties
+     * @since 9.88.5
+     */
+    public void setProperties(ConfluenceInputProperties properties)
+    {
+        this.properties.set(properties);
+    }
+
+    /**
+     * Checks whether a given space is forbidden to be overwritten based on the configured forbidden spaces.
+     * @param spaceKey the spaceKey of the space that is checked
+     * @return true if the specified space is listed among the forbidden spaces and cannot be overwritten
+     * @since 9.88.5
+     */
+    public boolean checkIfTheSpaceOverwriteIsForbidden(String spaceKey)
```

Comment:
```suggestion
    /**
     * @return whether it is fobbidden to overwrite a given space.
     * @param spaceKey the spaceKey of the space that is checked
     * @since 9.89.0
     */
    public boolean isSpaceOverwriteProtected(String spaceKey)
```

Some simplification of the method name, the javadoc (fewer repetitions).

You can't introduce new methods in a patch version, we will need to increase the minor version number.